### PR TITLE
Jest tests for Notification Services

### LINKED ISSUE #224: Setup jest for notification service and write some basic tests
### Expected Behavior

Jest is setup similar to other packages.
Running `yarn test` runs the tests
Some basic tests are set up to cover one area and provide an example

### Current Behavior

No jest, no tests

### PR DESCRIPTION
**Pull request by sallyjyl**
### Description
Set up jest tests for notification services in the src/blockscout/transfer.

### Tested
Passed all tests.

### Related issues

- Fixes #224

### Backwards compatibility
Yes

### CODE REVIEW

File: packages/notification-service/package.json [hunk 1]
```diff
@@ -11,12 +11,13 @@
...
     "gcp-build": "npm run build",
-    "test": "",
+    "test": "jest",
     "lint": "tslint -c tslint.json --project tsconfig.json",
     "lint-checks": "yarn run lint && yarn build --noEmit",
     "start": "node ./dist/index.js",
     "start:local": "npm run build; cp config/config.local.env .env; node --inspect ./dist/index.js",
-    "deploy": "./deploy.sh"
+    "deploy": "./deploy.sh",
+    "test:watch": "export TZ=UTC && jest --watch"
```
Comment (jmrossy) [inline packages/notification-service/package.json]: Please move this up to line 15.
Also suggest setting the TX on the `test` job and then making this line `yarn test --watch`

File: packages/notification-service/package.json [hunk 2]
```diff
@@ -37,8 +38,11 @@
     "@types/dotenv": "^4.0.3",
     "@types/express": "^4.16.1",
     "@types/i18next": "^11.9.3",
+    "@types/jest": "^24.0.16",
     "@types/node-fetch": "^2.1.2",
     "@types/utf8": "^2.1.6",
-    "@types/web3": "^1.0.18"
+    "@types/web3": "^1.0.18",
+    "jest": "^24.8.0",
+    "ts-jest": "^24.0.2"
```
Comment (jmrossy) [inline packages/notification-service/package.json]: Please use the same versions as the other projects (see mobile package.json). Difference dep versions across our packages can cause unexpected issues

File: packages/notification-service/test/transfers.test.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+jest.mock('../src/config', () => ({
+  BLOCKSCOUT_API: '',
+  STABLE_TOKEN_ADDRESS: '',
+  GOLD_TOKEN_ADDRESS: '',
+}))
+
+jest.mock('node-fetch', () => ({
+  ...jest.requireActual('node-fetch'),
```
Comment (jmrossy) [inline packages/notification-service/test/transfers.test.ts:42]: Is this line necessary?

File: packages/notification-service/test/transfers.test.ts [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+    expect(concated).toEqual([TRANSFER1, TRANSFER2])
+  })
+
+  it('should notify for new transfers since last block notified', async () => {
+    const transfers = [TRANSFER1, TRANSFER2]
+    const lastBlockNotified = 122
+    const returned = await notifyForNewTransfers(transfers, lastBlockNotified)
+
+    expect(sendPaymentNotificationMock).toHaveBeenCalled()
```
Comment (jmrossy) [inline packages/notification-service/test/transfers.test.ts]: Would be better to use `toHaveBeenCalledWith` here and specify the actual expected input
This test is a bit too permissive

File: packages/notification-service/test/transfers.test.ts [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    expect(returned.length).toEqual(transfers.length)
+  })
+
+  it('should skip for transfers older than last block notified', async () => {
+    const transfers = [TRANSFER1, TRANSFER2]
+    const lastBlockNotified = 130
+    const returned = await notifyForNewTransfers(transfers, lastBlockNotified)
+
+    expect(sendPaymentNotificationMock).toHaveBeenCalled()
```
Comment (jmrossy) [inline packages/notification-service/test/transfers.test.ts]: ditto

**Review by jmrossy — COMMENTED** [review]
Mostly looks good! Just a few comments. Also would be good to test deploy this to integration to confirm GCP supports es2017 since you changed the target output

File: packages/notification-service/test/transfers.test.ts (hunk 3 above)
Comment (sallyjyl) [reply packages/notification-service/test/transfers.test.ts:42]: It's needed for the last test in testing 'should update the last set block number'

**Review by jmrossy — DISMISSED** [review]
Nice!

File: packages/notification-service/package.json [hunk 6]
```diff
@@ -43,6 +43,8 @@
     "@types/utf8": "^2.1.6",
     "@types/web3": "^1.0.18",
     "jest": "^24.8.0",
-    "ts-jest": "^24.0.0"
+    "ts-jest": "^24.0.0",
+    "ts-node": "^7.0.1",
+    "typescript": "^3.3.3"
```
Comment (jmrossy) [inline packages/notification-service/package.json:47]: Yeah this should have been down here, please remove line 30 from the dependencies section

File: packages/notification-service/.gitignore [hunk 7]
```diff
@@ -1,5 +1,4 @@
 # Exclude compiled .js files
-*.js
```
Comment (jmrossy) [inline packages/notification-service/.gitignore]: Please restore this and add an exception just for jest.config.js

**Review by jmrossy — APPROVED** [review]

**Pull request merged**